### PR TITLE
Run GPU tests with PyTorch 1.12

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Use PyTorch 1.12 in Azure GPU CI

Discussed with @Borda 

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @carmocca @akihironitta @borda @tchaton @rohitgr7

### CODE REVIEW

**Review by Borda — APPROVED** [review]
you need first publish the docker image :)

**Comment by Borda** [comment]
@carmocca the GPU CI with PT 1.12 shall be up and running...

File: requirements/pytorch/check-avail-strategies.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import bagua  # noqa: F401
+# import bagua
```
Comment (carmocca) [inline requirements/pytorch/check-avail-strategies.py:1]: Is this comment on purpose?

File: .azure/gpu-tests.yml [hunk 2]
```diff
@@ -69,7 +69,9 @@ jobs:
       condition: eq(variables['continue'], '1')
 
     - bash: |
+        set -e
         python -c "fname = 'requirements/pytorch/strategies.txt' ; lines = [line for line in open(fname).readlines() if 'horovod' not in line] ; open(fname, 'w').writelines(lines)"
+        python -c "fname = 'requirements/pytorch/strategies.txt' ; lines = [line for line in open(fname).readlines() if 'bagua' not in line] ; open(fname, 'w').writelines(lines)"
         CUDA_VERSION_MM=$(python -c "import torch ; print(''.join(map(str, torch.version.cuda.split('.')[:2])))")
         pip install "bagua-cuda$CUDA_VERSION_MM>=0.9.0"
```
Comment (Borda) [inline .azure/gpu-tests.yml:76]: We need to install the spec CUDA version...

File: requirements/pytorch/check-avail-strategies.py (hunk 1 above)
Comment (Borda) [reply requirements/pytorch/check-avail-strategies.py:1]: good catch :)
```suggestion
import bagua
```

File: requirements/pytorch/check-avail-strategies.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import bagua  # noqa: F401
+import bagua
```
Comment (Borda) [inline requirements/pytorch/check-avail-strategies.py:1]:
```suggestion
import bagua # noqa: F401
```

**Review by awaelchli — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]
